github-actions/deploy-previews/pack-and-upload-artifact/action.yml

```diff
@@ -0,0 +1,49 @@
+name: Pack and upload Artifacts to Workflow
+author: 'Angular'
+description: |
+  Action that takes a built artifact, injects metadata, zips it up, uploads
+  it to the current GitHub action workflow.
+
+  Following best-practices from:
+  https://securitylab.github.com/research/github-actions-preventing-pwn-requests
+
+# NOTE: All inputs here are considered unsafe since this action is expected
+# to run in workflows from forked builds (allowing for arbitrary changes).
+inputs:
+  workflow-artifact-name:
+    required: true
+    description: |
+      Name of the artifact that should be deployed. A workflow may contain
+      multiple artifacts but only a single one can be picked for deployment.
+
+  pull-number:
+    required: true
+    description: Pull request for which the artifact is built for.
+
+  artifact-build-revision:
```

Comment:
Is there a compelling reason to not just have this required to be the Git SHA?

Reply:
No strong reason. In most cases it's the SHA, but it's just named build-revision, could have also been `build-sha`.

Reply:
Mostly by using the sha, we can determine exactly what the state of the repo was when the build occurred, thinking of "attacks" on it, it is valuable to just automatically use the sha value.

Reply:
I think there are two things:

1. I wouldn't want to automatically use the SHA here because the action should not assume/require a `pull_request` event type. There may be cases where we want to trigger this action on other triggers.
2. The SHA is not super significant because anybody could change this value to something else (e.g. an attacker). And in the case of previews for non-PRs a SHA might also not be available/easy to retrieve.

I'm happy to change this to `build-sha` but automatically using it from the event payload, and hence assuming it- I would prefer to avoid

Reply:
ahh gotcha, that works for me